### PR TITLE
Setup Branded Token and Utility Branded Token

### LINKED ISSUE #5: BrandedToken Facilitator agent
## Story
As a user
I want to facilitate branded token stake and redeem
Because it should surface errors in branded tokens
And it should provide feedback on how to improve the interaction layers

## Scenario
Given that I run the `bt` binary
When I pass it a sub-command
Then it executes the given sub-command and exits

## Info
The binary would have the following sub-commands:

Command | Explanation
--- | ---
`deploy` | Deploys a new gateway composer
`stake` | Stakes an amount into the branded token
`redeem` | Redeems an amount out of the branded token
`transpose` | Progresses tokens across chains, should be able to re-use most of the functionality of the `facilitate` binary (#3)

* `deploy` should deploy a new instance of the gateway composer and return the addresses to the command line
* `stake` and `redeem` should stake and redeem a given amount
* `transpose` should facilitate the stake/redeem processes

❓Do we need `transpose` as a command here or would that be taken over by the `facilitate` command from #3? How would the transpose command work exactly? (see examples)
❓How to use addresses? Extra addresses per bt? How funded?

Examples:
```
$ bt deploy
$ bt stake 20.00
$ bt redeem 1.5
$ bt transpose stake <?amount?>
$ bt transpose redeem <?amount?>
```

* Using the gateway composer and doing the staking must be done from two different addresses
* All logging must be done to a file (filename `bt_yearmonthdayhourminutesecond_randomstring.log`)
* Use [winston](https://github.com/winstonjs/winston) for logging
* Use [commander](https://github.com/tj/commander.js) like here: https://github.com/OpenSTFoundation/mosaic.js/pull/45

⚠️ Make sure that you align with other CLI tasks on commander and logging structure.
⚠️ Chain access and contract addresses should be read from the config file from #1.

### PR DESCRIPTION
**Pull request by sarvesh-ost**
This PR implements BT agent which setup Branded Token and Utility Branded Token. 

Partially fixes #5 

ℹ️ I am currently pointing to `brandedtoken.js` using git branch.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
   "scripts": {},
   "dependencies": {
     "@openstfoundation/mosaic.js": "^0.10.0-beta.4",
+    "@openstfoundation/brandedtoken.js": "https://github.com/sarvesh-ost/brandedtoken.js.git#Branded_token_setup",
```
Comment (benjaminbollen) [inline package.json]: 😇😂

File: src/config/chain_config.js [hunk 2]
```diff
@@ -14,15 +14,17 @@ class ChainConfig {
     this.auxiliaryChainId = process.env.ORIGIN_CHAIN_ID || 200;
     this.simpleTokenAddress = process.env.SIMPLE_TOKEN_ADDRESS || '0xca954C91BE676cBC4D5Ab5F624b37402E5f0d957';
     this.originMasterKey = process.env.ORIGIN_MASTER_KEY || '0x6d4e5f5ca54b88fdb68a5d0e6ea5aa3869f88116';
-    this.auxiliaryMasterKey = process.env.AUXILIARY_MASTER_KEY || '0xe84e1244e8c74b5b2df317d3ecf2f8ffd1f134d8';
+    this.auxiliaryMasterKey = process.env.AUXILIARY_MASTER_KEY || '0x490cc731b076e58509e1f7cee6c68b39cfe5ad62';
```
Comment (benjaminbollen) [inline src/config/chain_config.js]: we should remove these default keys

File: README.md [hunk 3]
```diff
@@ -140,3 +141,60 @@ node src/bin/ost.js unwrap <config.json> <address> <amount>
...
+
+## Setup Branded Token
+
+Prerequisite: `eip20TokenAddress`in config file. Optionally, you can run EIP20 Token setup step mentioned above. 
+
+```bash
+# Help: 
+node src/bin/bt.js --help
+
+node src/bin/bt.js brandedToken <config.json> <symbol> <name> <decimal> <conversionRate> <conversionDecimal>
```
Comment (schemar) [inline README.md]: It's actually called `setupBrandedToken` @sarvesh-ost

File: src/facilitator.js [hunk 4]
```diff
@@ -105,17 +105,92 @@ class Facilitator {
...
+      txOptions,
+    );
+    const nextNonce = await coGatewayInstance.contract.methods.getNonce(redeemer).call();
+    const currentNonce = parseInt(nextNonce, 10) - 1;
+
+    const activeProcess = await coGatewayInstance.contract.methods.getOutboxActiveProcess(
+      redeemer,
+    ).call();
+
+    const messageHash = activeProcess.messageHash_;
```
Comment (schemar) [inline src/facilitator.js]: Should be returned by `facilitator.redeem()` once mosaic.js is updated.

File: src/facilitator.js [hunk 5]
```diff
@@ -105,17 +105,92 @@ class Facilitator {
...
+
     logger.info('Redeem initiated');
+    return { messageHash, hashLock };
+  }
+
+  async progressRedeem(messageHash) {
+    logger.info('Redeem progress started');
+    const redeemRequest = this.chainConfig.redeems[messageHash];
+
+    // logger.info('redeemrequest :- ', redeemRequest);
```
Comment (schemar) [inline src/facilitator.js]: we should probably use international colon style (`foo: bar`) instead of Indian style (`foo :- bar`).

**Pull request merged**